horace_core/sqw/@axes_block/private/interpolate_data_.m

```diff
@@ -0,0 +1,129 @@
+function varargout = interpolate_data_(obj,nout,ref_nodes,density, ...
+    ref_grid_cell_size,proj)
+% interpolate density data for signal, error and number of
+% pixels provided as input density and defined on the references
+% nodes onto the grid, defined by this block
+%
+% Inputs:
+% nout      -- number of elements in cellarray of densities
+% ref_nodes -- 4D array of the nodes of the reference lattice,
+%              produced by get_density routine of the reference
+%              axes block and projected into coordinate system of this axes
+%              block
+% density   -- 3-elements cellarray containing arrays of
+%              signal, error and npix densities,
+%              produced by get_density routine of the reference
+%              axes block.
+% Optional:
+% ref_grid_cell_size
+%           -- 4D array of the scales of the reference lattice
+%              if missing or empty, assume ref_nodes have the same
+%              cell sizes as these nodes
+% proj      -- the projection object defining the transformation
+%              from this coordinate system to the system,
+%              where the reference nodes are defined
+%              If missing or empty, assume that this coordinate
+%              system and reference coordinate system are the
+%              same
+% Returns:
+% s,e,npix  -- interpolated arrays of signal, error and number
+%              of pixels calculated in the centres of the
+%              cells of this lattice.
+
+for i= 1:nargout
+    varargout{i} = [];
+end
```

Comment:
Is this necessary? Isn't this default?

Reply:
no, it will throw `variable not asigned` if not all present argouts are assigned